torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -82,12 +90,22 @@ std::shared_ptr<NCCLComm> NCCLComm::split(
       ncclCommSplit(sourceComm, color_id, rank, &(comm->ncclComm_), &config),
       sourceComm, // wait on parent comm
       std::nullopt);
+  if (color_id >= 0) {
+    // Waiting for parent comm above still does not seem to guarantee the child
+    // comm ptr is valid. Therefore we add a manual wait here for safety.
+    // TODO: remove this wait after NCCL fix the semantics.
+    while (!comm->ncclComm_) {
+      C10D_SCHED_SLEEP();
```

Comment:
hm, possible infinite sleep?

Reply:
Good call. Can I defer a jail break to the PR on top where we have a definite timeout value?
Today the default value is -1, which means no timeout.
I agree the deferral is not nice, but there is quite some work to pull part of the PR on top down here.

Reply:
Addressed in https://github.com/pytorch/pytorch/pull/138374